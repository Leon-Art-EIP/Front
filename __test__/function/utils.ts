import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
