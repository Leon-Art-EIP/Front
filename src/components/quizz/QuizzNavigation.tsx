import { Badge } from "../lib/lib";

export interface QuizzNavigationProps {
  questionIndex: number;
  questionsLength: number;
  setQuestionIndex: (index: number) => void;
  handlePreviousQuestion: () => void;
  handleNextQuestion: () => void;
}

export default function QuizzNavigation(props: QuizzNavigationProps): JSX.Element {
  return (
    <div className="flex items-baseline justify-center flex-row gap-24 w-full fixed bottom-14 select-none">
      <button
        onClick={props.handlePreviousQuestion}
        className="py-3 px-16 rounded-[30px] shadow-lg bg-[#E11C0A] text-white mt-10 hover:bg-[#c51708] disabled:bg-gray-300"
        disabled={props.questionIndex === 0}
        name="previous"
      >
        <span className="cursor-pointer">Précédente</span>
      </button>
      <span className="cursor-default text-2xl">
        {props.questionIndex + 1}/{props.questionsLength}
      </span>
      <button
        onClick={props.handleNextQuestion}
        className="py-3 px-16 rounded-[30px] shadow-lg bg-[#E11C0A] text-white mt-10 hover:bg-[#c51708] disabled:bg-gray-300"
        disabled={props.questionIndex === props.questionsLength - 1}
        name="previous"
      >
        <span className="cursor-auto">Suivante</span>
      </button>
    </div>
  );
}
