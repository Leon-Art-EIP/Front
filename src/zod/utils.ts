export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
