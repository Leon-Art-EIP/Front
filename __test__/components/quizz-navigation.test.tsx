import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import QuizzNavigation from "../../src/components/quizz/QuizzNavigation";

test("QuizzNavigation snapshot", () => {
  testSnapshot(
    <QuizzNavigation
      handleNextQuestion={() => {}}
      handlePreviousQuestion={() => {}}
      questionIndex={1}
      questionsLength={3}
    />
  );
});
