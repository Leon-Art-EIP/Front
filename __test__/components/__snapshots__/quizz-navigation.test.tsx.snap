// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuizzNavigation snapshot 1`] = `
<div
  className="flex items-baseline justify-center p-6 md:p-2 gap-8 md:gap-14 xl:gap-24 select-none"
>
  <button
    className="py-3 px-16 rounded-[30px] shadow-lg bg-[#E11C0A] text-white mt-10 hover:bg-[#c51708] disabled:bg-gray-300"
    name="previous"
    onClick={[Function]}
  >
    <span
      className="cursor-pointer"
    >
      Précédente
    </span>
  </button>
  <span
    className="cursor-default text-2xl"
  >
    2
    /
    3
  </span>
  <button
    className="py-3 px-16 rounded-[30px] shadow-lg bg-[#E11C0A] text-white mt-10 hover:bg-[#c51708] disabled:bg-gray-300"
    name="next"
    onClick={[Function]}
  >
    <span
      className="cursor-auto"
    >
      Suivante
    </span>
  </button>
</div>
`;
